executor/builder.go

```diff
@@ -656,12 +660,84 @@ func (b *executorBuilder) buildRevoke(revoke *ast.RevokeStmt) Executor {
 }
 
 func (b *executorBuilder) buildDDL(v *plannercore.DDL) Executor {
-	e := &DDLExec{
+	if b.ctx.GetSessionVars().CreateTableInsertingID != 0 {
+		// in a 'inserting data from select' state of creating table.
+		return b.buildCreateTableInsert(v, b.ctx.GetSessionVars().CreateTableInsertingID)
+	}
+	return &DDLExec{
 		baseExecutor: newBaseExecutor(b.ctx, v.Schema(), v.ExplainID()),
 		stmt:         v.Statement,
 		is:           b.is,
 	}
-	return e
+}
+
+// buildCreateTableInsert builds a CreateTableInsertExec to insert data when creating table by 'create table ... select'
+func (b *executorBuilder) buildCreateTableInsert(v *plannercore.DDL, tableID int64) Executor {
+	stmt, ok := v.Statement.(*ast.CreateTableStmt)
+	if !ok || v.InsertPlan.SelectPlan == nil {
+		b.err = errors.Errorf("Unexpected plan: %s", v.Statement.Text())
+		return nil
+	}
+	selectExec := b.build(v.InsertPlan.SelectPlan)
+
+	dom := domain.GetDomain(b.ctx)
+	ver, err := dom.Store().CurrentVersion()
+	if err != nil {
+		b.err = errors.Trace(err)
+		return nil
+	}
+	snapshot, err := dom.Store().GetSnapshot(ver)
+	if err != nil {
+		b.err = errors.Trace(err)
+		return nil
+	}
+
+	m := meta.NewSnapshotMeta(snapshot)
+	dbInfo, ok := dom.InfoSchema().SchemaByName(stmt.Table.Schema)
+	if !ok {
+		b.err = errors.Trace(err)
```

Comment:
`err` will be nil here.
b.err = infoschema.ErrDatabaseNotExists.GenWithStackByArgs(stmt.Table.Schema.L)

Reply:
adressed, thanks